docs/dev/migrating-to-v1.rst

```diff
@@ -1,79 +1,96 @@
-+==========================+======================================+
-| hydromt.config           | Removed                              |
-+--------------------------+--------------------------------------+
-| hydromt.log              | Removed (private: hydromt._utils.log)|
-+--------------------------+--------------------------------------+
-| hydromt.flw              | hydromt.gis.flw                      |
-+--------------------------+--------------------------------------+
-| hydromt.gis_utils        | hydromt.gis.utils                    |
-+--------------------------+--------------------------------------+
-| hydromt.raster           | hydromt.gis.raster                   |
-+--------------------------+--------------------------------------+
-| hydromt.vector           | hydromt.gis.vector                   |
-+--------------------------+--------------------------------------+
-| hydromt.gis_utils        | hydromt.gis.utils                    |
-+--------------------------+--------------------------------------+
+**Rationale**
 
+Given the changes to the `Model` class and new `ModelComponent` the `yaml` configuration file format has
+also been updated. The new format is more flexible and allows configuration of the model object and its
+components. Components can be added during instantiation by defining them in the yaml which is useful to
+define a model on-the-fly based on existing components. Note that this is however not necessary for plugins
+as the available components are defined in their implementation of the `Model` class.
 
-Changes to the format of the yaml interface
--------------------------------------------
+**Changes required**
 
 The first change to the YAML format is that now, at the root of the documents are three keys:
-`modeltype`, `global` and `steps`.
-- `modeltype` (optional) details what kind of model is going to be used in the model. This can currently also be provided only through the CLI,
-but given that YAML files are very model specific we've decided to make this available through the YAML file as well.
-- `global` is intended for any configuration for the model object itself, here you may override any default
-configuration for the components provided by your implementation. Any options mentioned here will be passed to the `Model.__init__` function
-- `steps` should contain a list of function calls. In pre-v1 versions this used to be a dictionary, but now it has become a list
-which removes the necessity for adding numbers to the end of function calls of the same name. You may prefix a component name
-for the step in a dotted manner to indicate the function should be called on that component instead of the model. In general any step
-listed here will correspond to a function on either the model or one of its components. Any keys that are listed under a step will be
-provided to the function call as arguments.
+
+- `modeltype` (optional) details what kind of model is going to be used in the model.
```

Comment:
@hboisgon @deltamarnix In my opinion `model_name` would be better. A type refers to a category of models, but here you want the actual model software name right?

```suggestion
- `model_name` (optional) the name of the model software.
```

Reply:
In addition, I think it would be nice to have a `Model.from_yml` class method to move the CLI functionality to the model class. In that way also python users can benefit from the yaml format. Something for v1-beta?